packages/snaps-sdk/src/types/handlers/name-lookup.ts

```diff
@@ -15,23 +15,45 @@ export type DomainLookupArgs = BaseOnNameLookupArgs & {
 };
 
 /**
- * The result of a domain lookup.
+ * The arguments for an address lookup.
+ *
+ * @property address - The address that is to be resolved.
+ */
+export type AddressLookupArgs = BaseOnNameLookupArgs & {
+  address: string;
+  domain?: never;
+};
+
+/**
+ * The address resolution object.
  *
+ * @property protocol - The protocol that resolved the domain.
  * @property resolvedAddress - The resolved address.
  */
-export type DomainLookupResult = {
+type AddressResolution = {
+  protocol: string;
   resolvedAddress: string;
-  resolvedDomain?: never;
 };
 
 /**
- * The arguments for an address lookup.
+ * The domain resolution object.
  *
- * @property address - The address that is to be resolved.
+ * @property protocol - The protocol that resolved the address.
+ * @property resolvedDomain - The resolved domain.
  */
-export type AddressLookupArgs = BaseOnNameLookupArgs & {
-  address: string;
-  domain?: never;
+type DomainResolution = {
```

Comment:
Should these be exported? 🤔

Reply:
Yeah, doesn't hurt. Might be useful in the extension somewhere.